std/algorithm/iteration.d

```diff
@@ -2969,10 +2969,20 @@ iterated from the back to the front, the separator will still be consumed from
 front to back, even if it is a bidirectional range too.
  */
 auto joiner(RoR, Separator)(RoR r, Separator sep)
-if (isInputRange!RoR && isInputRange!(ElementType!RoR)
-        && isForwardRange!Separator
-        && is(ElementType!Separator : ElementType!(ElementType!RoR)))
 {
+    static assert(isInputRange!RoR, "The range passed to 'joiner' does not "
+            , "fulfill isInputRange!", RoR.stringof);
```

Comment:
"does not fulfill" sounds a bit pedantic to me. Maybe just say "The range passed to joiner is not an InputRange"?

Reply:
fixed, thank you